l3kernel/l3file.dtx

```diff
@@ -3355,6 +3370,24 @@
 % \end{macro}
 % \end{macro}
 %
+% \begin{macro}{\file_input_raw:n, \@@_input:n}
+%   No error checking, no tracking.
+%    \begin{macrocode}
+\cs_new:Npn \file_input_raw:n #1
+  { \exp_args:Ne \@@_input_raw:n { \file_full_name:n {#1} } }
+\cs_new:Npx \@@_input_raw:n #1
+  {
+    \exp_not:N \tl_if_blank:nF {#1}
+      {
+        \exp_not:N \tex_input:D
+          \sys_if_engine_luatex:TF
```

Comment:
you can run only in one engine for a given run, so why make this a runtime test?

Reply:
This is all inside an `x`-type definition precisely for that reason :)